### PR TITLE
Section76 sectors and employee experience

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.QnA.Application.UnitTests/ServiceTests/NotRequiredProcessorPagesWithDoesNotContainTests.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        [TestCase("OrgType2", "OrgType1", true, false, "not present so should be removed")]
+        [TestCase("OrgType1", "OrgType2", true, false, "not present so should be removed")]
+        [TestCase("OrgType1", "OrgType1", false, false, "present so should not be removed")]
+        [TestCase("rgType1", "OrgType1", true, false, "not present so should be removed")]
+        [TestCase("OrgType", "OrgType1", true, false, "not present so should be removed")]
+        [TestCase("orgType1", "OrgType1", false, false, "present so should not be removed")]
+
+      
+
+        public void When_PagesWithNotRequired_DoesNotContain_conditions_are_removed(string notRequiredConditionValue, string applicationDataValue, bool toRemove, bool singleValue, string explanation)
```
Comment (David-B-Read) [inline src/SFA.DAS.QnA.Application.UnitTests/ServiceTests/NotRequiredProcessorPagesWithDoesNotContainTests.cs]: Is a [Test] needed here?

Comment (MarkFCain) [reply src/SFA.DAS.QnA.Application.UnitTests/ServiceTests/NotRequiredProcessorPagesWithDoesNotContainTests.cs]: Yes, will raise story for tests

Comment (MarkFCain) [reply src/SFA.DAS.QnA.Application.UnitTests/ServiceTests/NotRequiredProcessorPagesWithDoesNotContainTests.cs]: there were testcase there, but I left 3 spaces by mistake....

File: src/SFA.DAS.QnA.Application/Commands/AnswerValidator.cs [hunk 2]
```diff
@@ -26,12 +26,15 @@ public AnswerValidator(IValidatorFactory validatorFactory)
 
                 if (question.Input.Options == null) continue;
 
-                foreach (var option in question.Input.Options.Where(option => answerToThisQuestion?.Value == option.Value && option.FurtherQuestions != null))
+                foreach (var option in question.Input.Options.Where(option => answerToThisQuestion?.Value != null && option.FurtherQuestions != null))
                 {
-                    foreach (var furtherQuestion in option.FurtherQuestions)
+                    if (answerToThisQuestion?.Value != null && answerToThisQuestion.Value.Contains(option.Value))
```
Comment (martin-hornsby) [inline src/SFA.DAS.QnA.Application/Commands/AnswerValidator.cs]: Could shorten this down as you check  answerToThisQuestion?.Value != null above

File: src/SFA.DAS.QnA.Application/Services/NotRequiredProcessor.cs [hunk 3]
```diff
@@ -41,6 +41,30 @@ public IEnumerable<Page> PagesWithoutNotRequired(List<Page> pages, JObject appli
...
+                    var fieldValue = applicationData[fieldToCheck]?.Value<string>();
+                    if (!string.IsNullOrEmpty(fieldValue))
+                    {
+
+                        var applicationDataValues = fieldValue.Split(",", StringSplitOptions.RemoveEmptyEntries);
+                        foreach (var adv in applicationDataValues)
+                        {
+                            foreach (var condition in notRequiredCondition.DoesNotContain)
+                            {
+                                if (string.Equals(adv, condition, StringComparison.CurrentCultureIgnoreCase)) isPresent = true;
```
Comment (martin-hornsby) [inline src/SFA.DAS.QnA.Application/Services/NotRequiredProcessor.cs]: should be InvariantCultureIgnoreCase to be consistent across QnA

**Pull request merged**